server.cpp

```diff
@@ -42,36 +47,117 @@ size_t read_complete(const char* buff, const error_code& err, size_t bytes)
 
 void write_log(const std::string& outfile, const std::string& log_message)
 {
+    const std::string log_msg = make_daytime_string() + " " + log_message;
```

Comment:
You may receive `log_message` by value and just modify (prepend with date) it. No need for he new variable.

Reply:
Even better - you can out them separately:
```c++
std::cout << make_daytime_string() << " " << log_message << "\n";
```